src/wallet/migrate.cpp

```diff
@@ -1,13 +1,547 @@
+        s >> re_len;
+        s >> re_pad;
+        s >> root;
+        s >> unused3;
+        s >> crypto_magic;
+        s >> trash;
+        s >> iv;
+        s >> chksum;
+    }
+};
+
+/** General class for records in a BDB BTree database. Contains common fields. */
+class RecordHeader
+{
+public:
+    uint16_t len;  // Key/data item length
+    RecordType type;  // Page type and DELETE FLAG
+
+    static constexpr size_t SIZE = 3; // The record header is 3 bytes
+
+    RecordType GetRealType() const
+    {
+        return static_cast<RecordType>(static_cast<uint8_t>(type) & ~static_cast<uint8_t>(RecordType::DELETE));
+    }
+
+    bool IsDeleted() const
+    {
+        return static_cast<uint8_t>(type) & static_cast<uint8_t>(RecordType::DELETE);
+    }
+
+    template<typename Stream>
+    void Unserialize(Stream& s)
+    {
+        s >> len;
+
+        uint8_t uint8_type;
+        s >> uint8_type;
+        type = static_cast<RecordType>(uint8_type);
+    }
+};
+
+/** Class for data in the record directly */
+class DataRecord
+{
+public:
+    DataRecord(const RecordHeader& header) : m_header(header) {}
+    DataRecord() = delete;
+
+    RecordHeader m_header;
+
+    std::vector<std::byte> data; // Variable length key/data item
+
+    template<typename Stream>
+    void Unserialize(Stream& s)
+    {
+        data.resize(m_header.len);
+        s.read(AsWritableBytes(Span(data.data(), data.size())));
+    }
+};
+
+/** Class for records representing internal nodes of the BTree. */
+class InternalRecord
+{
+public:
+    InternalRecord(const RecordHeader& header) : m_header(header) {}
+    InternalRecord() = delete;
+
+    RecordHeader m_header;
+
+    uint8_t unused;                   // Padding, unused
+    uint32_t page_num;                // Page number of referenced page
+    uint32_t records;                 // Subtree record count
+    std::vector<std::byte> data;  // Variable length key item
+
+    static constexpr size_t FIXED_SIZE = 9; // Size of fixed data is 9 bytes
+
+    template<typename Stream>
+    void Unserialize(Stream& s)
+    {
+        s >> unused;
+        s >> page_num;
+        s >> records;
+
+        data.resize(m_header.len);
+        s.read(AsWritableBytes(Span(data.data(), data.size())));
+    }
+};
+
+/** Class for records representing overflow records of the BTree.
+ * Overflow records point to a page which contains the data in the record.
+ * Those pages may point to further pages with the rest of the data if it does not fit
+ * in one page */
+class OverflowRecord
+{
+public:
+    OverflowRecord(const RecordHeader& header) : m_header(header) {}
+    OverflowRecord() = delete;
+
+    RecordHeader m_header;
+
+    uint8_t unused2;      // Padding, unused
+    uint32_t page_number; // Page number where data begins
+    uint32_t item_len;    // Total length of item
+
+    static constexpr size_t SIZE = 9; // Overflow record is always 9 bytes
+
+    std::vector<std::byte> data; // Data from all of the overflow pages
```

Comment:
In commit "wallet: implement independent BDB deserializer in BerkeleyRODatabase" d5fe130106bdf367c0bca157b02d12254590b585

Can this `data` field be removed? Seems to be unused.

Reply:
Removed. I think originally I though I could put the overflow page data into this field, but I didn't end up doing it that way.